sdks/java/extensions/protobuf/src/test/java/org/apache/beam/sdk/extensions/protobuf/ProtoCoderTest.java

```diff
@@ -167,4 +169,20 @@ public void testNonDeterministicProperty() throws CoderException {
     Coder<MessageWithMap> coder = ProtoCoder.of(MessageWithMap.class);
     assertNotEquals(CoderUtils.encodeToBase64(coder, msg2), CoderUtils.encodeToBase64(coder, msg1));
   }
+
+  @Test
+  public void testDynamicMessage() throws Exception {
+    DynamicMessage message =
+        DynamicMessage.newBuilder(MessageA.getDescriptor())
+            .setField(
+                MessageA.getDescriptor().findFieldByNumber(MessageA.FIELD1_FIELD_NUMBER), "foo")
+            .build();
+    Coder<DynamicMessage> coder = ProtoCoder.of(message.getDescriptorForType());
```

Comment:
This is the type I would expect `ProtoCoder.of(descriptor)`. If it really has some type magic where it can be `Coder<T>` instead of `Coder<DynamicMessage>` then I suggest a test of that functionality.

Reply:
It can't. I'll change the `of(DynamicMessage` to return Coder<DynamicMessage> as you suggested.